src/event/writer.rs

```diff
@@ -161,9 +173,46 @@ impl EventWriter {
                 .expect("send error");
             rx_error
         } else {
+            self.event_handles.push_back(rx_flush);
             rx
         }
     }
+
+    /// Flush data.
+    ///
+    /// It will wait until all pending appends have acknowledgment.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// let mut byte_writer = client_factory.create_byte_writer(segment).await;
+    /// let payload = vec![0; 8];
+    /// let size = event_writer.write_event(&payload).await;
+    /// event_writer.flush().await;
+    /// ```
+    pub async fn flush(&mut self) -> Result<(), Error> {
+        while self.event_handles.front().is_some() {
+            let handle = self.event_handles.pop_front().expect("get first handle");
+            let event_result = handle.await.map_err(|e| Error::InternalFailure {
+                msg: format!("oneshot error {:?}", e),
+            })?;
+            event_result?;
+        }
+        Ok(())
+    }
+
+    /// This is to check if all the events have been flushed.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// let mut byte_writer = client_factory.create_byte_writer(segment).await;
+    /// let payload = vec![0; 8];
+    /// let size = event_writer.write_event(&payload).await;
+    /// event_writer.flush().await;
+    /// assert!(event_writer.flush_cleared());
+    ///   ```
+    pub fn flush_cleared(&self) -> bool {
+        self.event_handles.is_empty()
```

Comment:
This doesn't handle the case where the are event_handles, but they are completed.

Reply:
I think it is better to just take this out, because the flush() is already checking all handles and trying to use receiver to tell if it has completed or not. The other option is to track the senders for each oneshot, which doesn't seem optimal.